geoserver/src/eumetsat/pinning/src/main/java/org/geoserver/eumetsat/pinning/views/ViewsClient.java

```diff
@@ -0,0 +1,106 @@
+/* (c) 2025 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.eumetsat.pinning.views;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.time.Instant;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.eumetsat.pinning.config.PinningServiceConfig;
+import org.geotools.util.logging.Logging;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+/**
+ * Client for fetching and parsing map views from a preferences endpoint.
+ *
+ * <p>This class handles retrieving view data via REST API and converting view preferences into a
+ * format usable by the GeoServer EUMETSAT pinning service.
+ */
+public class ViewsClient {
+
+    private static final Logger LOGGER = Logging.getLogger(ViewsClient.class);
+
+    private final RestTemplate restTemplate;
+    private final ObjectMapper objectMapper;
+    private final Catalog catalog;
+    private final String baseApiUrl;
+
+    @Autowired private final PinningServiceConfig config;
+
+    private static final String SEARCH_PATH = "search/findByType?type=mapView";
+    private static final String LAST_UPDATE_FILTER = "&lastUpdate=?";
+
+    public ViewsClient(Catalog catalog, PinningServiceConfig config) {
+        this.catalog = catalog;
+        this.config = config;
+        this.baseApiUrl = config.preferencesUrl();
+        this.restTemplate = new RestTemplate();
+        this.objectMapper = new ObjectMapper();
+    }
+
+    /**
+     * Fetches map views from the preferences endpoint, optionally filtered by last update time.
+     *
+     * @param lastUpdatedFilter Optional timestamp to filter views by their last update time
+     * @return A list of View objects retrieved from the preferences endpoint
+     * @throws Exception If there are issues with REST request or JSON parsing
+     */
+    public List<View> fetchViews(String lastUpdatedFilter) throws Exception {
```

Comment:
It's not wrong per se, but it's odd that a Java API dealing with dates takes an externally formatted string... normally one would get an Instant and format it locally?